util/importers/ldc93s1.py

```diff
@@ -74,14 +80,14 @@ def _compute_source_target(self):
 
     def _populate_batch_queue(self, session):
         source, source_len, target, target_len = self._compute_source_target()
-        while True:
+        while not self._coord.should_stop():
             try:
                 session.run(self._enqueue_op, feed_dict={
                     self._x: source,
                     self._x_length: source_len,
                     self._y: target,
                     self._y_length: target_len})
-            except (RuntimeError, tf.errors.CancelledError):
+            except Exception as e:
```

Comment:
Yeah it's to general. I can switch it.

However, it should not be switched it back to how it was. The old code was also in error too.

We should **not** have caught `RuntimeError`. `RuntimeError` is thrown when we close a session, then call `run()` on it. We should never close a session then call `run()` on it as then the session is then in invalid state. Catching `RuntimeError` here was hiding part of the cause of the crashing bug.

Catching `tf.errors.CancelledError` seems reasonable, but I am not sure if this is the only thing we need to catch.

I can switch it, then re-run the traning-validation transition 50 more times to see if a problem occurs.